core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -854,7 +866,15 @@ public void onFailure(Throwable error) {
       if (!isIdempotent || error instanceof FrameTooLongException) {
         decision = RetryDecision.RETHROW;
       } else {
-        decision = retryPolicy.onRequestAborted(statement, error, retryCount);
+        try {
+          decision = retryPolicy.onRequestAborted(statement, error, retryCount);
+        } catch (Throwable cause) {
+          setFinalError(
+              new IllegalStateException("Unexpected error while invoking the retry policy", cause),
+              null,
+              execution);
```

Comment:
Other invocations of the retry policy are already protected by similar try/catch blocks. This one would propagate directly to `InFlightHandler`, I think it's better to handle it here.